Pontoon/tbx_merge/tbx_merge.py

```diff
@@ -0,0 +1,176 @@
+import os
+import argparse
+import requests
+from pathlib import Path
+from xml.etree import ElementTree as et
+
+
+class hashabledict(dict):
+    def __hash__(self):
+        return hash(tuple(sorted(self.items())))
+
+
+class XMLCombiner(object):
+    def __init__(self, filenames):
+        if len(filenames) == 0:
+            raise FileNotFoundError("Invalid path, or path contains no valid files.")
+
+        self.roots = [et.parse(f).getroot() for f in filenames]
+
+    def combine(self):
+        for r in self.roots[1:]:
+            self.combine_element(self.roots[0], r)
+        return et.ElementTree(self.roots[0])
+
+    def combine_element(self, one, other):
+        """
+        This function recursively combines text, attributes, and children of an xml element tree.
+        It updates either the text/attributes/children of an element if another element is found in `one`,
+        or adds it from `other` if not found.
+        """
+        mapping = {(el.tag, hashabledict(el.attrib)): el for el in one}
+        for el in other:
+            if len(el) == 0:
+                # Not nested
+                try:
+                    mapping[(el.tag, hashabledict(el.attrib))].text = el.text
+                except KeyError:
+                    # Element not found in the mapping
+                    mapping[(el.tag, hashabledict(el.attrib))] = el
+                    one.append(el)
+            else:
+                # Nested
+                try:
+                    # Recursively process the element, and update it in the same way
+                    self.combine_element(mapping[(el.tag, hashabledict(el.attrib))], el)
+                except KeyError:
+                    # Element not found in the mapping
+                    mapping[(el.tag, hashabledict(el.attrib))] = el
+                    one.append(el)
+
+
+def extract_smartling_id_term(f):
+    """ "Creates a dictionary with (term, definition) as key, and the Smartling UID as value."""
+    root = et.parse(f).getroot()
+    smartling_dict = {}
+    for termEntry in root.iter("termEntry"):
+        term = termEntry.find(
+            "./langSet[@{http://www.w3.org/XML/1998/namespace}lang='en-US']/tig/term"
+        ).text
+        try:
+            definition = termEntry.find("./descrip[@type='definition']").text
+        except AttributeError:
+            definition = None
+        id = termEntry.attrib["id"]
+        smartling_dict[(term, definition)] = id
+
+    return smartling_dict
+
+
+def replace_pontoon_ids(etree, smartling_dict):
+    """Replaces Pontoon IDs with Smartling IDs if the term and definition match exactly a term in Smartling glossary file."""
+    root = etree.getroot()
+    pontoon_term_map = {}
```

Comment:
nit: it's a `map` here, but it's a `dict` in `smartling_dict`, should probably be the same.